### DISCUSSION TITLE
Was getting O365 logs, now I'm not

### DISCUSSION THREAD

**Discussion by gbigger** [Security-Onion-Solutions/securityonion#7378] [Unsupported Versions] [answered]
Not sure what happened, but for some reason in the last 2 weeks we are no longer getting any logs from O365.   I ran this test and it said completed successfully, and I am not seeing any obvious reason for this in logs, so any help would be greatly appreciated.

-- so-elastalert-test :

Choose a rule to test from the list above (must be typed exactly as shown above): email-o365_internation-userloginfailed-success.yml
The results can be rather long.  Would you like to write the results to a file? (y/N) n

Didn't get any results.
1 rules loaded

Would have written the following documents to writeback index (default is elastalert_status):

elastalert_status - {'rule_name': 'O365 Alert - International UserLoginFailed Success', 'endtime': datetime.datetime(2022, 3, 1, 22, 24, 42, 312211, tzinfo=tzutc()), 'starttime': datetime.datetime(2022, 2, 28, 22, 24, 42, 312211, tzinfo=tzutc()), 'matches': 0, 'hits': 0, '@timestamp': datetime.datetime(2022, 3, 1, 22, 24, 55, 541519, tzinfo=tzutc()), 'time_taken': 12.72232985496521}

Test completed successfully!

**Comment by dougburks** [comment, marked as answer]
Are you not getting alerts from ElastAlert or are you not getting any logs from O365 at all? Can you manually search in Hunt or Kibana and find current O365 logs?

**Reply by gbigger** [reply]
we are getting alerts from ElastAlerts. and it does look like we are getting some logs from event.module o365.  But it does not look like any of the dashboards are populating those logs.

**Reply by dougburks** [reply]
Are you able to provide screenshots of where you are seeing the data and where you expect to see the data but aren't?

**Reply by gbigger** [reply]
![O365 Login's](https://user-images.githubusercontent.com/42288299/157128173-7e761b35-3d26-4d8e-8c0c-5c7f0f1c955b.png)
![O365 Dashboard](https://user-images.githubusercontent.com/42288299/157128449-cbcb62c8-bccf-408a-8765-1a27a913d2bb.png)
It looks like there are some O365 logins on the Login's png, but the dashboard isn't populating any O365 failed logins.  I can't tell if it is an issue getting logs from O365 or an issue with the dashboard populating the logs.

**Reply by dougburks** [reply]
In your second screenshot, what happens if you make the query just:
```
event.dataset:"o365.audit"
```
and then remove the filter underneath that?

**Reply by gbigger** [reply]
![o365 audit](https://user-images.githubusercontent.com/42288299/157537675-d31a24c3-fe01-494e-9248-8e1d26cc6617.png)

It did not populate any data :-(

**Reply by dougburks** [reply]
You might want to investigate the shard failures by clicking the `Show details` button.

You might also try checking the logs in /opt/so/log/.

**Reply by gbigger** [reply]
what is the best way to test the authentication process to o365?  my guess is something is wrong with the login process to o365 and that is where the issue is.

**Reply by dougburks** [reply]
If filebeat is failing to login to o365, then I would expect filebeat to log something to that effect in its log file.

**Reply by gbigger** [reply]
ok, I ran the soup command on SO and rebooted all the servers, seems to be working now.  Is there a way to build an alert that logs are not being processed or parsed?

**Reply by dougburks** [reply]
One option might be an ElastAlert flatline rule:
https://docs.securityonion.net/en/2.3/elastalert.html
https://elastalert2.readthedocs.io/en/latest/ruletypes.html#flatline
